standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1451,11 +1453,11 @@ public void commitTxn(CommitTxnRequest rqst) throws NoSuchTxnException, TxnAbort
                 OperationType.UPDATE + "," + OperationType.DELETE + ")";
         long tempCommitId = generateTemporaryId();
 
-        if (txnType.get() == TxnType.COMPACTION) {
+        if (txnType == TxnType.SOFT_DELETE || txnType == TxnType.COMPACTION) {
           acquireTxnLock(stmt, false);
           commitId = getHighWaterMark(stmt);
 
-        } else if (txnType.get() != TxnType.READ_ONLY && !isReplayedReplTxn) {
+        } else if (txnType != TxnType.READ_ONLY && !isReplayedReplTxn) {
```

Comment:
What will happen wrt replication of these soft delete-s? Do they know about the changes we propose?

Reply:
I need to check if we send the metadata delete event for repl, however, for the data removal, we are doing this in Cleaner.